code/game/objects/items/circuitboards/machine_circuitboards.dm

```diff
@@ -446,21 +444,21 @@
 	needs_anchored = FALSE
 	var/is_special_type = FALSE
 
-/obj/item/circuitboard/machine/smartfridge/apply_default_parts(obj/machinery/smartfridge/M)
-	build_path = M.base_build_path
+/obj/item/circuitboard/machine/smartfridge/apply_default_parts(obj/machinery/smartfridge/smartfridge)
+	build_path = smartfridge.base_build_path
 	if(!fridges_name_paths.Find(build_path, fridges_name_paths))
-		name = "[initial(M.name)] (Machine Board)" //if it's a unique type, give it a unique name.
+		name = "[initial(smartfridge.name)] (Machine Board)" //if it's a unique type, give it a unique name.
 		is_special_type = TRUE
 	return ..()
 
-/obj/item/circuitboard/machine/smartfridge/attackby(obj/item/I, mob/user, params)
-	if(!is_special_type && I.tool_behaviour == TOOL_SCREWDRIVER)
-		var/position = fridges_name_paths.Find(build_path, fridges_name_paths)
-		position = (position == fridges_name_paths.len) ? 1 : (position + 1)
-		build_path = fridges_name_paths[position]
-		to_chat(user, span_notice("You set the board to [fridges_name_paths[build_path]]."))
-	else
-		return ..()
+/obj/item/circuitboard/machine/smartfridge/screwdriver_act(mob/living/user, obj/item/tool)
+	. = TRUE
```

Comment:
Change to `return TRUE`, at the end, and likely `return FALSE` at the `is_special_type` check.